experiments/service.go

```diff
@@ -0,0 +1,99 @@
+package experiments
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	logutil "github.com/argoproj/argo-rollouts/utils/log"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+var experimentKind = v1alpha1.SchemeGroupVersion.WithKind("Experiment")
+
+func (c *Controller) getServicesForExperiment(experiment *v1alpha1.Experiment) (map[string]*corev1.Service, error) {
+	svcList, err := c.serviceLister.Services(experiment.Namespace).List(labels.Everything())
```

Comment:
will it throw error if there are no services..as in current flow, if create service is false, then we. don't create services...so do we need to check for the flag and return nil?